src/Microsoft.AspNet.Http.Extensions/HeaderDictionaryTypeExtensions.cs

```diff
@@ -112,11 +113,11 @@ internal static IList<T> GetList<T>([NotNull] this IHeaderDictionary headers, st
             if (KnownListParsers.TryGetValue(typeof(T), out temp))
             {
                 var func = (Func<IList<string>, IList<T>>)temp;
-                return func(headers.GetValues(name));
+                return func(headers[name]);
             }
 
-            var values = headers.GetValues(name);
-            if (values == null || !values.Any())
+            var values = headers[name];
+            if (StringValues.IsNullOrEmpty(values))
```

Comment:
I think this usage shows some of the benefits of the API. Checking for null or !any is pretty annoying.